api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3968,8 +4208,8 @@ public void putObject(String bucketName, String objectName, InputStream stream,
    * @param partNumber
    *          Part number of multipart put object.
    */
-  private String putObject(String bucketName, String objectName, int length,
-      Object data, String uploadId, int partNumber, Map<String, String> headerMap)
+  private String putObject(String bucketName, String objectName, Object data, int length,
```

Comment:
change `int length` to `long length`

Reply:
Why is this change needed?

Reply:
because precision is lost i.e. `long` is passed as `int`.  Depending on the platform int and long sizes are different.

Reply:
As discussed int is apt here